vendor/processors/authorizenet.py

```diff
@@ -406,132 +409,98 @@ def get_transaction_raw_response(self):
 
-        )
+        if transaction_data:
+            for key, value in transaction_data.__dict__.items():
+                if value.text:
+                    response_data.update({key: value.text})
 
-    def parse_success(self, subscription=True):
-        self.transaction_submitted = False
-        if self.transaction_response:
-            if not subscription:
-                if self.transaction_response.messages.resultCode == "Ok":
-                    if hasattr(self.transaction_response.transactionResponse, 'messages') is True:
-                        self.transaction_submitted = True
-            else:
-                if self.transaction_response.messages.resultCode == "Ok":
-                    self.transaction_submitted = True
+        if 'messages' in self.transaction_response.__dict__:
+            response_data['resultCode'] = self.transaction_response.messages.resultCode.text
+            response_data['messages'] = {key: value.text for key, value in self.transaction_response.messages.message.__dict__.items() if value.text}
 
+        return response_data
 
-    def save_payment_subscription(self):
+    def parse_response(self, parser_function):
         """
-        Processes the transaction reponse from the gateway so it can be saved in the payment model
-        The transaction id is the subscription id returned by Authorize.Net
+        Processes the transaction response from the gateway so it can be saved in the payment model
         """
-        self.transaction_id = self.transaction_message.get("subscription_id", 'failed_payement')
+        parser_function()
 
-        self.transaction_response = self.make_transaction_response(
-            raw=self.get_transaction_raw_response(),
-            messages=f'trans id is {self.transaction_id}'
-        )
+    def is_transaction_response_empty(self):
 
-    def save_subscription_result(self):
-        """
-        Processes the transaction reponse from the gateway so it can be saved in the payment model
-        The transaction id is the subscription id returned by Authorize.Net
-        """
-        self.transaction_id = self.transaction_message.get("subscription_id", 'failed_subscription')
+        if not self.transaction_response:
+            self.set_transaction_info(
+                raw="No transaction Response was set",
+                errors="No transaction Response was set",
+                messages="No transaction Response was set"
+            )
+            return True
+        
+        return False
+
+    def parse_payment_response(self):
+        self.transaction_info = {}
+
+        if self.is_transaction_response_empty():
+            return None
 
-        self.transaction_response = self.make_transaction_response(
+        errors = self.get_transaction_errors(self.transaction_response.transactionResponse)
+        
+        if not hasattr(self.transaction_response, 'transactionResponse'):
```

Comment:
Since you're checking for `transactionResponse` here, probably want to move line 470 below the check as well to avoid errors